packages/dashboard-core-plugins/src/panels/ChartPanel.tsx

```diff
@@ -271,10 +272,16 @@ export class ChartPanel extends Component<ChartPanelProps, ChartPanelState> {
     }
 
     if (settings !== prevState.settings && isFigureChartModel(model)) {
-      model.updateSettings(settings);
       this.updatePanelState();
     }
 
+    if (
+      ChartSettings !== prevProps.metadata.settings &&
```

Comment:
Not sure where this block came from, this isn't right - `prevProps.metadata.settings` is never going to change after the component is created.
The `model.updateSettings(settings)` should still be in the previous block, and this block should be deleted.

Ah - also, the `settings` in `ChartPanelState` and `PanelState` should both be `ChartModelSettings` type, not `WorkspaceSettings`. Only the `settings` in `ChartPanelProps` should be `WorkspaceSettings` type.

The prop is the workspace settings being set on the panel and is not modifiable, whereas the state settings is for the chart itself and can be modified.